packages/audioplayers/tizen/src/audioplayers_tizen_plugin.cc

```diff
@@ -167,80 +229,98 @@ class AudioplayersTizenPlugin : public flutter::Plugin {
     if (iter != audio_players_.end()) {
       return iter->second.get();
     }
+    return nullptr;
+  }
+
+  void CreateAudioPlayer(const std::string &player_id) {
+    auto event_channel = std::make_unique<FlEventChannel>(
+        registrar_->messenger(), "xyz.luan/audioplayers/events/" + player_id,
+        &flutter::StandardMethodCodec::GetInstance());
+    event_sinks_[player_id];
+    OnSetEventSink on_set_event_sink =
+        [this, id = player_id](std::unique_ptr<FlEventSink> event_sink) {
+          this->event_sinks_[id] = std::move(event_sink);
+        };
 
-    PreparedListener prepared_listener = [channel = channel_.get()](
-                                             const std::string &player_id,
-                                             int32_t duration) {
-      flutter::EncodableMap wrapped = {{flutter::EncodableValue("playerId"),
-                                        flutter::EncodableValue(player_id)},
-                                       {flutter::EncodableValue("value"),
-                                        flutter::EncodableValue(duration)}};
-      auto arguments = std::make_unique<flutter::EncodableValue>(wrapped);
-      channel->InvokeMethod("audio.onDuration", std::move(arguments));
+    event_channel->SetStreamHandler(
+        std::make_unique<AudioPlayerStreamHandler>(on_set_event_sink));
+    event_channels_[player_id] = std::move(event_channel);
```

Comment:
The reason for using `event_channels_` is to maintain the lifecycle of unique_ptr? Is it used elsewhere?

Reply:
Deleted.